kb/development/deprecation.md

```diff
@@ -0,0 +1,30 @@
+# Deprecations
+
+To ensure backwards-compatibility between versions of Mbed TLS, functions in the public interface may not be removed except during the release of a new major version. Instead, functions may be 'deprecated', indicating to users that they will be removed in the future.
+
+To deprecate a function, the following actions must be taken:
```

Comment:
Missing subsection title for functions. Possibly there should be a separate subsection for common stuff like `platform_util.h` and `MBEDTLS_TEST_DEPRECATED`.

Reply:
Done, duplicating `platform_util.h` and `MBEDTLS_TEST_DEPRECATED` as it seemed to flow better.